src/Nethermind/Nethermind.Consensus/IBlockProducer.cs

```diff
@@ -1,16 +1,19 @@
 // SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
-using System;
+using System.Threading;
 using System.Threading.Tasks;
+using Nethermind.Consensus.Producers;
 using Nethermind.Core;
+using Nethermind.Evm.Tracing;
 
 namespace Nethermind.Consensus;
 
 public interface IBlockProducer
 {
-    Task Start();
-    Task StopAsync();
-    bool IsProducingBlocks(ulong? maxProducingInterval);
-    event EventHandler<BlockEventArgs> BlockProduced;
+    Task<Block?> BuildBlock(
+        BlockHeader? parentHeader = null,
+        CancellationToken? cancellationToken = null,
```

Comment:
minor: convention puts cancellationToken often as last parameter to separate it from more logic-oriented parameters.